homeassistant/components/onvif/device.py

```diff
@@ -165,17 +165,13 @@ async def async_check_date_and_time(self) -> None:
                 )
                 return
 
+            tzone = dt_util.DEFAULT_TIME_ZONE
+            cdate = device_time.LocalDateTime
```

Comment:
Per the [ONVIF Core Specification](https://www.onvif.org/specs/core/ONVIF-Core-Specification.pdf) **8.3.6 GetSystemDateAndTime**: `LocalDateTime` is optional, so should not be used as a default here.

The only provision made is this:
> A device shall provide the UTCDateTime information although the item is marked as optional to ensure backward compatibility.

That being said (since we treated it as an either/or previously) I would be OK just moving this back to the `else` condition instead of default to ensure we don't break cameras that do not provide a `LocalDateTime` and only provide a `UTCDateTime` object.

Reply:
The current code (prior to this PR) does access `device_time.LocalDateTime` if `UTCDateTime` is not defined, even though `LocalDateTime` might not be defined. In this case, `cdate` would then be `None`, which would be handled by the `if cdate is None` right below.

In the new code, if a camera does not provide a `LocalDateTime`, `cdate` will initially be `None`, and will be set to `UTCDateTime` if it's defined. Why would it break cameras without a `LocalDateTime`, then?

To me, adding an `else` branch would only result in more code for the exact same behavior, unless I am missing something? 😊

Reply:
I forgot that Zeep generates the model off of the WSDL and not the response body. In that case, you are correct that property should always exist and default to `None`. My initial thought was `UTCDateTime` should always be set according to the spec, but `LocalDateTime` may not be so the property may not exist on the dynamic object and throw an exception.

That being said, it *may* be considered more readable if the logic was contained similarly within the `if/else` as we are defaulting the TZ to something not provided in the response, but not `cdate`.  It would at least prevent people like me from not fully understanding what's going on in the first read (even if it is technically the same).

That's a matter of opinion though, so I'll unblock this for now.